test/test_cuda.py

```diff
@@ -1437,6 +1437,41 @@ def test_streams(self):
         default_stream.synchronize()
         self.assertTrue(default_stream.query())
 
+    @unittest.skipIf(not TEST_MULTIGPU, "detected only one GPU")
+    @skipIfRocm
+    def test_stream_context(self):
```

Comment:
I would prefer a more targeted test-case here for the stream context behavior. For example, this implicitly tests that the device and stream switch by using query() and _sleep. I'd rather just check that the device and stream are set appropriately and reset appropriately. For example, something like:

- Switch to s1, check that torch.cuda.current_stream() is s1 and current_device() is 1
- Check that stream and device are restored after context manager
- Check nested context managers